### PR TITLE
HS-1413: simplify exporter parameter and add includeOther

### PR DESCRIPTION
**Pull request by freddy61025**
## Jira link(s)
- https://opennms.atlassian.net/browse/HS-1413

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts

### CODE REVIEW

File: rest-server/src/test/java/org/opennms/horizon/server/service/flows/FLowClientTest.java [hunk 1]
```diff
@@ -78,21 +69,18 @@ public class FLowClientTest {
...
+    private final boolean includeOther = true;
     private final Instant startTime = Instant.now();
     private final Instant endTime = startTime.minus(1, ChronoUnit.HOURS);
 
-
     @Before
     public void startGrpc() throws IOException {
-//        private static ApplicationsServiceGrpc.ApplicationsServiceImplBase applicationsServiceBlockingStub;
-//        private static ExporterServiceGrpc.ExporterServiceImplBase exporterServiceStub;
         var applicationsServiceBlockingStub = mock(ApplicationsServiceGrpc.ApplicationsServiceImplBase.class, delegatesTo(
```
Comment (FrancescoBuc86) [inline rest-server/src/test/java/org/opennms/horizon/server/service/flows/FLowClientTest.java:84]: these comments can be removed?

File: rest-server/src/main/java/org/opennms/horizon/server/service/flows/FlowClient.java [hunk 2]
```diff
@@ -132,15 +150,19 @@ public Series getApplicationSeries(RequestCriteria requestCriteria, String tenan
...
         if (requestCriteria.getExporter() != null) {
-            requestCriteria.getExporter().stream().forEach(e ->
-                seriesRequest.addFilters(convertExporter(e))
-            );
+            requestCriteria.getExporter().forEach(e ->
+            {
+                var filter = convertExporter(e);
+                if (filter != null) {
+                    seriesRequest.addFilters(filter);
+                }
```
Comment (FrancescoBuc86) [inline rest-server/src/main/java/org/opennms/horizon/server/service/flows/FlowClient.java:164]: Optional.ofNullable(convertExporter(e)).ifPresent(filter -> seriesRequest.addFilters(filter));

File: rest-server/src/main/java/org/opennms/horizon/server/service/flows/FlowClient.java [hunk 3]
```diff
@@ -106,15 +117,22 @@ public Summaries getApplicationSummaries(RequestCriteria requestCriteria, String
...
 
         if (requestCriteria.getExporter() != null) {
-            requestCriteria.getExporter().stream().forEach(e ->
-                summaryRequest.addFilters(convertExporter(e))
+            requestCriteria.getExporter().forEach(e ->
+                {
+                    var filter = convertExporter(e);
+                    if (filter != null) {
+                        summaryRequest.addFilters(filter);
+                    }
```
Comment (FrancescoBuc86) [inline rest-server/src/main/java/org/opennms/horizon/server/service/flows/FlowClient.java:134]: requestCriteria.getExporter().forEach(exporter -> Optional.ofNullable(convertExporter(exporter)).ifPresent(summaryRequest::addFilters));

File: rest-server/src/main/java/org/opennms/horizon/server/service/flows/FlowClient.java [hunk 4]
```diff
@@ -69,30 +68,42 @@ public void shutdown() {
...
         }
         if (requestCriteria.getExporter() != null) {
-            requestCriteria.getExporter().stream().forEach(e ->
-                listRequest.addFilters(convertExporter(e))
+            requestCriteria.getExporter().forEach(e ->
+                {
+                    var filter = convertExporter(e);
+                    if (filter != null) {
+                        listRequest.addFilters(filter);
+                    }
```
Comment (FrancescoBuc86) [inline rest-server/src/main/java/org/opennms/horizon/server/service/flows/FlowClient.java:89]: requestCriteria.getExporter().forEach(exporter -> Optional.ofNullable(convertExporter(exporter)).ifPresent(listRequest::addFilters));

File: rest-server/src/test/java/org/opennms/horizon/server/service/flows/FLowClientTest.java (hunk 1 above)
Comment (freddy61025) [reply rest-server/src/test/java/org/opennms/horizon/server/service/flows/FLowClientTest.java:84]: it just left over from last time.

**Review by FrancescoBuc86 — APPROVED** [review]

**Pull request merged**